example/http/client/sync/http_client_sync_post.cpp

```diff
@@ -0,0 +1,199 @@
+//
+// Copyright (c) 2021-2022 Guilherme Schvarcz Franco (guilhermefrancosi at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+//------------------------------------------------------------------------------
+//
+// Example: HTTP client, synchronous, POST with payload data
+//
+//------------------------------------------------------------------------------
+
+//[example_http_client_post
+
+#include <boost/asio/connect.hpp>
+#include <boost/asio/ip/tcp.hpp>
+#include <boost/beast/core.hpp>
+#include <boost/beast/http.hpp>
+#include <boost/beast/version.hpp>
+#include <boost/filesystem/path.hpp>
+#include <boost/json.hpp>
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+namespace beast = boost::beast;     // from <boost/beast.hpp>
+namespace http = beast::http;       // from <boost/beast/http.hpp>
+namespace net = boost::asio;        // from <boost/asio.hpp>
+using tcp = net::ip::tcp;           // from <boost/asio/ip/tcp.hpp>
+
+struct parameters {
+    explicit parameters(int argc, char** argv);
+    http::message_generator make_request() const;
+
+    char const* host() const { return args_.at(1); }
+    char const* port() const { return args_.at(2); }
+private:
+    char const* program_name() const { return args_.at(0); }
+    char const* target() const { return args_.at(3); }
+    int version() const { return !strcmp("1.0", args_.at(5)) ? 10 : 11; }
+
+    std::array<char const*, 6> args_{{
```

Comment:
What is all this `parameters`? I don't like introducing a bunch of new idioms..

Reply:
Yeah. I don't like very intransparent spaghetti around arvg[3] and argv[argc>3? 5 : 2]. Would you like to introduce Boost Program Options instead :)

Personally, I dislike that the cookbook-style examples take command line arguments at all. It just distracts from what is being demonstrated. At least with the parameters helper you can ignore the cruft when you read the program.

For servers, perhaps a docroot, maybe. Though current working directory would be fine for me.

Reply:
I like the examples the way they are written, which introduce as few new idioms as possible. There is no confusion or things to learn here, because it is straight C++ with no user defined types:
https://github.com/boostorg/beast/blob/00293a6adb5383fe14217a8916dd7ff79a857483/example/advanced/server/advanced_server.cpp#L612

Reply:
Yeah I got that, but to me the "it's all ints and char, what's the big deal" is a meager excuse. It would be 10x more readable with just hardcoded defaults. If that doesn't suit the user, they can choose to edit or indeed take from `argv` (since it is so trivial).

I really do not wish to complicate that command line, just putting more blocks on top of the jenga pile, so I'll mull on a better way to solve this again.